keras_cv/models/segmentation/deeplab.py

```diff
@@ -136,6 +150,15 @@ def __init__(
             )
         self.segmentation_head = segmentation_head
 
+        weights = parse_weights(weights, False, "deeplabv3")
+        if weights and not tf.io.gfile.exists(weights):
+            raise ValueError(
+                "The `weights` argument should be either `None`, the path to the "
+                "weights file to be loaded, or a reference to pre-trained weights. "
+                f"Weights file not found at location: {weights}"
+            )
+        self.load_weights(weights)
```

Comment:
interesting...does this work? Today all our models are subclassed models, meaning the variables are not created at this point, I wonder if `load_weights` will error out

Reply:
Ahh yes I forgot that we have to call `call` on a subclass model before this will load.

So perhaps we want to actually override `load_weights` for these models to support our custom weights?
For now, I've removed the `weights` param from this PR, as we should probably discuss how we'll offer pre-trained non-backbone models.

IIRC KerasNLP has come up with a `from_pretrained` API that we might want to consider?

Reply:
the `from_pretrained` doesn't help much on this case, because KerasNLP actually creates all the variables during init. I have an action item to make our models to be function subclass models though, after which adding `weights` would be easier.

Reply:
sg -- for now would you like me to override `load_weights`?

Reply:
it's ok not to override, I will make this complete before 0.4 anyway